### PR TITLE
Feature: add `echo` operator to `Socket`

### PR DESCRIPTION
**Pull request by tusharmath**

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/socket/Socket.scala [hunk 1]
```diff
@@ -57,8 +57,14 @@ sealed trait Socket[-R, +E, -A, +B] { self =>
 }
 
 object Socket {
+
   def collect[A]: PartialCollect[A] = new PartialCollect[A](())
 
+  /**
+   * Simply echos the incoming message back
+   */
+  def echo[A]: Socket[Any, Nothing, A, A] = Socket.collect[A] { case a => ZStream.succeed(a) }
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/socket/Socket.scala:66]: test missing

Comment (girdharshubham) [reply zio-http/src/main/scala/zhttp/socket/Socket.scala:66]: done.

File: zio-http/src/test/scala/zhttp/socket/SocketSpec.scala [hunk 2]
```diff
@@ -23,38 +25,46 @@ object SocketSpec extends DefaultRunnableSpec {
...
+        val socketC: Socket[Any, Nothing, Int, Int] = socketB.provide(42)
 
-      assertM(socketC.execute(1000).runCollect)(equalTo(Chunk(12)))
-    }
+        assertM(socketC.execute(1000).runCollect)(equalTo(Chunk(12)))
+      } +
+      testM("echo") {
+        val stream = Socket.echo(WebSocketFrame.text("FOO"))
+
+        assertM(stream.runCollect)(equalTo(Chunk(WebSocketFrame.text("FOO"))))
```
Comment (amitksingh1490) [inline zio-http/src/test/scala/zhttp/socket/SocketSpec.scala]:
```suggestion
        assertM(Socket.echo(1).runCollect)(equalTo(Chunk(1)))
```

Comment (girdharshubham) [reply zio-http/src/test/scala/zhttp/socket/SocketSpec.scala]: done.

**Review by amitksingh1490 — APPROVED** [review]

**Pull request merged**